### PR TITLE
Fix for #2949.

### PR DESCRIPTION
**Pull request by osdf**
Extend the set of types that are not handled by json.dumps to elementary types.

*Issue #2949 

*Description of changes:*

*Testing done:*
Local test.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/test_estimator.py [hunk 1]
```diff
@@ -1268,7 +1267,7 @@ def test_custom_code_bucket_without_prefix(time, sagemaker_session):
     _, _, train_kwargs = sagemaker_session.train.mock_calls[0]
     assert train_kwargs["hyperparameters"]["sagemaker_submit_directory"] == json.dumps(
         expected_submit_dir
-    )
+    )
```
Comment (ahsan-z-khan) [inline tests/unit/test_estimator.py:1270]: lets remove this blank space

**Review by ahsan-z-khan — CHANGES_REQUESTED** [review]
`black-check` is failing. Can you reformat?

**Comment by mufaddal-rohawala** [comment]
Thanks for contributing, 
Unit tests have been failing for this PR, can you fix them @osdf

I would suggest fixing this by a change in [here](https://github.com/aws/sagemaker-training-toolkit/blob/master/src/sagemaker_training/logging_config.py#L29) to accept strings, rather than changing Estimator.

**Comment by navinsoni** [comment]
Closing as this change has been addressed by https://github.com/aws/sagemaker-python-sdk/pull/3043

**Pull request closed without merging**